src/test/test_proj_data_info_subsets.cxx

```diff
@@ -91,30 +91,42 @@ run_tests()
 void TestProjDataInfoSubsets::
 test_split(const ProjData &proj_data)
 {
-    std::vector<int> even_views;
-    std::vector<int> odd_views;
-
-    for (int view = 0; view < proj_data.get_num_views(); ++ view)
-      {
-        if (view % 2 == 0)
-            even_views.push_back(view);
-        else
-            odd_views.push_back(view);
-      }
+  int view;    
+  int num_subsets = 4;
+
+  std::vector<int> subset_views;
 
-    ProjDataInMemory even_subset = proj_data.get_subset(even_views);
-    ProjDataInMemory odd_subset = proj_data.get_subset(odd_views);
+  for (int subset = 0; subset < num_subsets; ++subset){
+    // create a vector containg every num_subsest-th view starting at subset
+    // for num_subsets = 4 and subset = 0 this is [0, 4, 8, 12, ...]
+    // for num_subsets = 4 and subset = 1 this is [1, 5, 9, 13, ...]
 
-    int segment = 0; // for segment
+    subset_views.clear();
+    view = subset;
 
-    // dodgy: assume n views is even
-    for (int subset_view = 0; subset_view < even_subset.get_num_views(); subset_view++)
-      {
-        check_if_equal(
-            proj_data.get_viewgram(even_views[subset_view], segment),
-            even_subset.get_viewgram(subset_view, segment),
-            "Are viewgrams equal?");
+    while(view < proj_data.get_num_views()){
+      subset_views.push_back(view);
+      view += num_subsets;
+    }
+
+    ProjDataInMemory subset_proj_data = proj_data.get_subset(subset_views);
+
+    // loop over views in the subset data and compare them against the original "full" data
+    for(std::size_t i = 0; i < subset_views.size(); ++i){
+      // i runs from 0, 1, ... views_in_subset - 1 and indicates the view number in the subset
+      // the corresponding view in the original data is at subset_views[i]
+
+      // TODO: Check how to get number of segments from proj_data
+      // loop over all segments to check viewgram for all segments
+      for (int segment = 0; segment < proj_data.get_num_segments(); ++segment){
+        // TODO: implement compare_views
+        compare_views(
+            proj_data.get_viewgram(subset_views[i], segment),
+            subset_proj_data.get_viewgram(i, segment), "Are viewgrams equal?");
+        // TODO also compare viewgram metadata
```

Comment:
somewhat tricky as it will be different. However, we should at least compare `viewgram.get_proj_data_info_sptr()->get_phi(Bin(....))`. Possibly this doesn't need to be from the `viewgram` but just use the original `proj_data_info`, as those are pretty much guaranteed to be the same (could add a check if you want)